### PR TITLE
[GLUTEN-8094][CH][Part-1] Support reading data from the iceberg with CH backend

### PR DESCRIPTION
**Pull request by zzcclp**
## What changes were proposed in this pull request?
Support reading data from the iceberg with CH backend

- basic iceberg scan transformer
- read from the iceberg with the copy-on-write mode

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: gluten-substrait/src/main/scala/org/apache/gluten/backendsapi/TransformerApi.scala [hunk 1]
```diff
@@ -76,4 +76,7 @@ trait TransformerApi {
   def invalidateSQLExecutionResource(executionId: String): Unit = {}
 
   def genWriteParameters(fileFormat: FileFormat, writeOptions: Map[String, String]): Any
+
+  /** use Hadoop Path class to encode the file path */
+  def encodeFilePathIfNeed(filePath: String): String = filePath
```
Comment (zhztheplayer) [inline gluten-substrait/src/main/scala/org/apache/gluten/backendsapi/TransformerApi.scala:81]: It turn out that a similar difference exists on regular scan

VL https://github.com/apache/incubator-gluten/blob/e56da333dc33de90540efa6dc63efa70c3c71ce8/backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxIteratorApi.scala#L124-L126
CH https://github.com/apache/incubator-gluten/blob/e56da333dc33de90540efa6dc63efa70c3c71ce8/backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHIteratorApi.scala#L166

Maybe the logics should be somehow consolidated in future, I am not sure.

**Review by zhztheplayer — APPROVED** [review]
+1 on the common change, thanks.

File: backends-clickhouse/src/test-iceberg/scala/org/apache/gluten/execution/iceberg/ClickHouseIcebergSuite.scala [hunk 2]
```diff
@@ -0,0 +1,640 @@
...
+ * limitations under the License.
+ */
+package org.apache.gluten.execution.iceberg
+
+import org.apache.gluten.execution.{GlutenClickHouseWholeStageTransformerSuite, IcebergScanTransformer}
+import org.apache.gluten.GlutenConfig
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.Row
+
+class ClickHouseIcebergSuite extends GlutenClickHouseWholeStageTransformerSuite {
```
Comment (liujiayi771) [inline backends-clickhouse/src/test-iceberg/scala/org/apache/gluten/execution/iceberg/ClickHouseIcebergSuite.scala:24]: Is it possible to allow CH and Velox to share this part of the test cases?

Comment (zzcclp) [reply backends-clickhouse/src/test-iceberg/scala/org/apache/gluten/execution/iceberg/ClickHouseIcebergSuite.scala:24]: will share this part after the CH backend support the merge on read mode for the iceberg, and there is also a bug when using the timestamp type as partition column for the CH backend.

**Pull request merged**